pkg/types/pipeline_runner.go

```diff
@@ -0,0 +1,26 @@
+package types
+
+import (
+	"context"
+	"time"
+)
+
+type Vars struct {
+	Vars map[string]interface{}
+}
+
+type Options struct {
+	MaxTaskDuration time.Duration
+}
+
+type TaskResult struct {
+	ID    string
+	Type  string
+	Value interface{}
```

Comment:
How do we send Vars and Values over the wire? Is there an appropriate interface we could swap in? e.g. `json.(Un)Marshaler`?

Reply:
Well I guess JSON is a bad example since it operates on `interface{}` :shrug:

Reply:
Vars will be translated to a `structpb.Struct` and Value will be translated to a `structpb.Value`. Those provide extensive support for interface{}

Reply:
Ah you mean: can we be more structured? Hmmm... let me have a think: I think we could make it more structured but that would require changing the pipeline runner itself (which would be worth it for the type safety).

Reply:
Oh I hadn't seen https://pkg.go.dev/google.golang.org/protobuf/types/known/structpb. I suppose this would simplify some things in chainreader, if we committed to JSON there.